### PR TITLE
PT-1134: Rework edit dictionary values in the Platform Settings

### PR DESCRIPTION
**Pull request by akak1977**
https://virtocommerce.atlassian.net/browse/PT-1131

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js [hunk 1]
```diff
@@ -37,12 +42,9 @@ angular.module('platformWebApp').controller('platformWebApp.settingDictionaryCon
...
     };
 
-    $scope.add = function (form) {
-        if (form.$valid) {
-            currentEntities.push($scope.newValue);
-            resetNewValue();
-            form.$setPristine();
-        }
+    $scope.filteredEntities = function () {
+        loverCasedSearchText = blade.searchText.toLowerCase();
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js:46]: 1. typo in lowerCasedSearchText
2. add var

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js [hunk 2]
```diff
@@ -109,32 +129,79 @@ angular.module('platformWebApp').controller('platformWebApp.settingDictionaryCon
...
+    $scope.inputKeyDown = function ($event) {
+        if ($event.keyCode === 13) {
+            // Apply value on hit Enter
+            $scope.applyValue();
+        };
+        if ($event.keyCode === 27) {
+            // Decline value on hit Esc
+            $scope.selectItem(null);
+        };
+    };
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js:170]: 1. can move to inline html?
2. can use ng-keyup?

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js [hunk 3]
```diff
@@ -109,32 +129,79 @@ angular.module('platformWebApp').controller('platformWebApp.settingDictionaryCon
...
+        orderedEntities = _.sortBy(entities, function (o) { return o.value; })
+        if (blade.orderDesc) {
+            orderedEntities = orderedEntities.reverse();
+        }
+        // Overkill to preserve array reference
+        entities.length = 0;        
+        for (const obj of orderedEntities) {
+            entities.push(obj);
+        }
+    }
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js:182]: rewrite using .sort

**Review by eMazeika — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js (hunk 2 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js:170]: 1. I tried before, but looks bad.
2. OK

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/settings/blades/setting-dictionary.js]: Thanks!

**Review by Vectorfield4 — APPROVED** [review]

**Review by eMazeika — APPROVED** [review]

**Pull request merged**